client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HttpConnection.ts

```diff
@@ -189,6 +189,17 @@ export class HttpConnection implements IConnection {
         return normalizedUrl;
     }
 
+    private resolveNegotiateUrl(url: string): string {
+        let index = url.indexOf("?");
+        let negotiateUrl = this.url.substring(0, index == -1 ? url.length : index);
+        if (negotiateUrl[negotiateUrl.length - 1] !== '/') {
+            negotiateUrl += '/';
+        }
+        negotiateUrl += "negotiate";
+        negotiateUrl += index == -1 ? "" : url.substring(index);
+        return negotiateUrl;
+    }
+
```

Comment:
Should we have tests for this?

Reply:
Probably

Reply:
Unfortunate that we have to do this :(. I like it better than splitting the query string off though. Looks fine, as long as it's well tested.